src/psyclone/transformations.py

```diff
@@ -225,10 +226,10 @@ def validate(self, node, options=None):
 
         # Check we are not a sequential loop
         # TODO add a list of loop types that are sequential
-        if node.loop_type == 'colours':
-            raise TransformationError("Error in "+self.name+" transformation. "
-                                      "The target loop is over colours and "
-                                      "must be computed serially.")
+        if isinstance(node, PSyLoop) and node.loop_type == 'colours':
```

Comment:
It's this kind of check that I'm wondering about for the null loop in LoopTrans.

Reply:
I reintroduced the null check, although I think this needs rethinking. Maybe if a loop is parallelisable should be logic in the loop class, e.g. loop.is_parallelisable . Internally the generic implementation will just use the dependency analysis but the LFRicLoop, GOLoop, NemoLoop can use additional loop_type checks to validate(or not) other certain loops.  I think this would be sensible because the logic to know if a loop is parallelisable must be repeated in a few transformation.
But I leave this for another PR.

Reply:
Yes, that makes sense.